pkg/config/teeconfig/teeconfig.go

```diff
@@ -111,91 +124,209 @@ func (t *teeConfig) ParseEnvAsSlice(key string, fn func(string) []interface{}) {
 
 // IsSet wraps Viper for concurrent access
 func (t *teeConfig) IsSet(key string) bool {
-	return t.baseline.IsSet(key)
+	base := t.baseline.IsSet(key)
+	compare := t.compare.IsSet(key)
+	if base != compare {
+		log.Warnf("difference in config: IsSet(%s) -> base: %v | compare %v", key, base, compare)
+	}
+	return base
 }
 
 func (t *teeConfig) AllKeysLowercased() []string {
-	return t.baseline.AllKeysLowercased()
+	base := t.baseline.AllKeysLowercased()
+	compare := t.compare.AllKeysLowercased()
+	slices.Sort(base)
+	slices.Sort(compare)
+	if !reflect.DeepEqual(base, compare) {
+		log.Warnf("difference in config: allkeyslowercased() -> base len: %d | compare len: %s", len(base), len(compare))
+		d := 0
+		for i := range base {
+			if i < len(compare) {
+				if base[i] == compare[d] {
+					d++
+				} else {
+					log.Warnf("AllKeysLowercased item %d *** : %s (%v) | %s", i, base[i], t.baseline.Get(base[i]), compare[d])
+				}
+			}
+			log.Flush()
+		}
```

Comment:
I think it would be better instead to rely on the fact that the keys are sorted, and compare them this way (pseudo-code):
```
i := 0
j := 0
for i < len(base) && j < len(compare) {
  if base[i] < compare[j] {
    log()
    i++
  } else if base[i] > compare[j] {
    log()
    j++
  } else {
    i++
    j++
  }
}
```